.github/scripts/audit-permissions.sh

```
@@ -0,0 +1,195 @@
+    fi
+    
+    # Check 4: Check for concurrency controls
+    if yq eval '.concurrency' "$workflow_file" | grep -q "null"; then
+        print_warn "No concurrency controls defined"
+        print_info "  Add 'concurrency:' to cancel outdated runs"
+    else
+        print_pass "Concurrency controls configured"
+    fi
+    
+    # Check 5: Check for pinned actions (not using @v1 tags)
+    local unpinned_actions=$(yq eval '.. | select(has("uses")) | .uses' "$workflow_file" 2>/dev/null | grep -v "^\./" | grep "@" || echo "")
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)